src/webServiceClient.ts

```diff
@@ -74,97 +77,94 @@ export default class WebServiceClient {
     const parsedPath = `/minfraud/v2.0/${path}`;
     const url = `https://${this.host}${parsedPath}`;
 
-    const options = {
-      auth: `${this.accountID}:${this.licenseKey}`,
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => controller.abort(), this.timeout);
+
+    const options: RequestInit = {
+      body: postData,
       headers: {
         Accept: 'application/json',
-        'Content-Length': Buffer.byteLength(postData),
+        Authorization:
+          'Basic ' +
+          Buffer.from(`${this.accountID}:${this.licenseKey}`).toString(
+            'base64'
+          ),
+        'Content-Length': Buffer.byteLength(postData).toString(),
         'Content-Type': 'application/json',
         'User-Agent': `minfraud-api-node/${version}`,
       },
-      host: this.host,
       method: 'POST',
-      path: parsedPath,
-      timeout: this.timeout,
+      signal: controller.signal,
     };
 
-    return new Promise((resolve, reject) => {
-      const req = https.request(options, (response) => {
-        let data = '';
-
-        response.on('data', (chunk) => {
-          data += chunk;
-        });
-
-        response.on('end', () => {
-          if (response.statusCode && response.statusCode === 204) {
-            return resolve();
-          }
-
-          try {
-            data = JSON.parse(data);
-          } catch {
-            return reject(this.handleError({}, response, url));
-          }
-
-          if (response.statusCode && response.statusCode !== 200) {
-            return reject(
-              this.handleError(data as ResponseError, response, url)
-            );
-          }
-
-          return resolve(new modelClass(data));
-        });
-      });
-
-      req.on('error', (err: NodeJS.ErrnoException) => {
-        return reject({
-          code: err.code,
-          error: err.message,
-          url,
-        } as WebServiceClientError);
-      });
-
-      req.write(postData);
-
-      req.end();
-    });
+    let data;
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        return Promise.reject(await this.handleError(response, url));
```

Comment:
I was a bit confused by this. I would have expected this to throw the exception directly (or even for `handleError` to do that.

I assume this is like this because you wanted to use the same catch for both the `await fetch` and the `data = await response.json()`. It would have been clearer to me to throw here and just rethrow instances of `WebServiceClientError` in the catch, but maybe the reasoning here is clear enough to most readers.

Reply:
I like how it separates handling "bad" server responses from client-side errors and aborts.  I created a separate branch with the suggested change; let me know if it's what you had in mind: https://github.com/maxmind/minfraud-api-node/commit/65cd8fe1a1bbede9eca38dd83524e38f018eb602

Reply:
Yeah, I guess that is what I had in mind, but I agree that it ends up being a bit messier than I would like. A comment or something might help people understand why it is like this.

Potentially, you could also move the `response.ok` check out of the try/catch too and pass the data into `handleError`, but that may be a bit more involved, especially for errors where the body isn't JSON.

Reply:
https://github.com/maxmind/minfraud-api-node/pull/1491/commits/3d1c9748566fb26bc5e9a9c4075a9d3d127ba4e0